server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2286,7 +2302,10 @@ public IndexCommitRef acquireLastIndexCommit(final boolean flushFirst) throws En
         // the to a write lock when we fail the engine in this operation
         if (flushFirst) {
             logger.trace("start flush for snapshot");
-            flush(false, true);
+            // TODO: Is it a concern to block here?
```

Comment:
I think we should split into two methods - or just let clients do a flush themselves first. This is fine though for this initial PR, though you might as well just use the default `flush(boolean, boolean)` method?